test/workers/delete_payment_setting_hierarchy_worker_test.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class DeletePaymentSettingHierarchyWorkerTest < ActiveSupport::TestCase
+  def setup
+    tenant = FactoryBot.create(:simple_provider)
+    tenant.schedule_for_deletion!
+    @payment_gateway_setting = FactoryBot.create(:payment_gateway_setting, account: tenant, tenant_id: tenant.id)
+    @buyers = FactoryBot.create_list(:simple_buyer, 2, provider_account: tenant)
+  end
+
+  attr_reader :payment_gateway_setting, :buyers
+
+  test 'it performs for the buyers is it is destroyed by the provider association' do
+    Sidekiq::Testing.inline! do
```

Comment:
since this is a activejob worker, what do you think about using:
```
perform_enqueued_jobs(only: DeletePaymentSettingHierarchyWorker)
```
?

Reply:
@duduribeiro i can do that but without the `only` because i want it to always execute the whole hierarchy chain, no matter which worker uses for the whole chain.